### PR TITLE
Moved to error stream

### PR DESCRIPTION
**Pull request by RamblingCookieMonster**
First stab at using pester.... Let me know if you have any preferences on organizing the tests, haven't worked with this before!

Moved from returning Streams.Error directly to looping through this collection and writing to the error stream.

@vors - I swapped out the Write-Warning when errors occur.  I see the value in separating errors within the runspace though, if you prefer that feedback mechanism feel free to swap it back in!

### CODE REVIEW

**Comment by vors** [comment]
LGTM

File: Invoke-Parallel.Tests.ps1 [hunk 1]
```diff
@@ -7,5 +7,11 @@ Describe 'Invoke-Parallel with default parameters' {
         $out.Count | Should Be 11
         $out[5][0] | Should Be 'a'
     }
+
+    It 'should output runspace errors to error stream' {
+        $out = 0 | Invoke-Parallel -ScriptBlock {Write-Error "A Fake Error!"} -ErrorVariable OutError
+        $out.Count | Should Be 0
+        $OutError[0].ToString() | Should Be "A Fake Error!"
```
Comment (vors) [inline Invoke-Parallel.Tests.ps1]: What is `$OutError[0]`? Should not it be `$Error[0]`?

**Comment by RamblingCookieMonster** [comment]
Used ErrorVariable OutError to redirect errors to that variable.  Shouldn't happen, but just in case $Error is contaminated by other errors.

**Comment by vors** [comment]
:+1:

**Pull request merged**